build/docker/zap_common.py

```diff
@@ -298,13 +298,20 @@ def zap_active_scan(zap, target, policy):
     logging.debug(zap.ascan.scan_progress(ascan_scan_id))
 
 
-def zap_wait_for_passive_scan(zap):
+def zap_wait_for_passive_scan(zap, timeout = 0):
     rtc = zap.pscan.records_to_scan
     logging.debug('Records to scan...')
+    time_taken = 0
     while (int(zap.pscan.records_to_scan) > 0):
         logging.debug('Records to passive scan : ' + zap.pscan.records_to_scan)
         time.sleep(2)
-    logging.debug('Passive scanning complete')
+        time_taken += 2
+        if timeout and time_taken > timeout:
+          break
+    if timeout and time_taken > timeout:
```

Comment:
Set a boolean "timedout = false" or something like that and check it for the completed message?